### PR TITLE
[DLS-7603]: partnership and llp trading name navigation

### PR DESCRIPTION
**Pull request by BackRowOla**

### CODE REVIEW

File: app/uk/gov/hmrc/fhregistrationfrontend/views/businessPartners/v2/business_partner_partnership_trading_partner_name.scala.html [hunk 1]
```diff
@@ -61,10 +61,9 @@
     ))
 }
 
-@layout(title = titlePrefix + title, backUrl = Some("#")) {
+@layout(title = titlePrefix + title, backUrl = None, isBrowserBackMimic = true) {
```
Comment (gjstirling) [inline app/uk/gov/hmrc/fhregistrationfrontend/views/businessPartners/v2/business_partner_partnership_trading_partner_name.scala.html:64]: Needs a back URL for scenario 1 ?

File: it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersLtdLiabilityPartnershipControllerISpec.scala [hunk 2]
```diff
@@ -9,20 +10,22 @@ class BusinessPartnersLtdLiabilityPartnershipControllerISpec
...
-      WsTestClient.withClient { client =>
-        val result = client.url(s"$baseUrl/form/business-partners/limited-liability-partnership-name")
-          .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie))
-          .get()
+        WsTestClient.withClient { client =>
+          val result = client.url(s"$baseUrl/form/business-partners/limited-liability-partnership-name")
+            .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie))
+            .withHttpHeaders(xSessionId,
+              "Csrf-Token" -> "nocheck")
+            .post(Map("ltdLiabilityPartnershipName" -> Seq("Partnership Name")))
```
Comment (gjstirling) [inline it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersLtdLiabilityPartnershipControllerISpec.scala:23]: Should this test be in the POST section ?

File: test/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnerPartnershipTradingNameControllerSpec.scala [hunk 3]
```diff
@@ -43,7 +44,7 @@ class BusinessPartnerPartnershipTradingNameControllerSpec extends ControllerSpec
         setupUserAction()
 
         when(mockAppConfig.newBusinessPartnerPagesEnabled).thenReturn(true)
-        val request = FakeRequest()
+        val request = FakeRequest().withCookies(Cookie(name = "businessType", value = ""))
```
Comment (gjstirling) [inline test/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnerPartnershipTradingNameControllerSpec.scala:47]: Add a business type ?

**Review by gjstirling — CHANGES_REQUESTED** [review]

File: test/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnerPartnershipTradingNameControllerSpec.scala (hunk 3 above)
Comment (BackRowOla) [reply test/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnerPartnershipTradingNameControllerSpec.scala:47]: Removed the cookie as its not needed here

File: it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersLtdLiabilityPartnershipControllerISpec.scala (hunk 2 above)
Comment (BackRowOla) [reply it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersLtdLiabilityPartnershipControllerISpec.scala:23]: Yes, this meant to be the post request, must have made the mistake when i rebasing. I've reverted it back to a GET.

**Pull request merged**